service/awsconfig/v10/resource/s3bucket/error.go

```diff
@@ -36,6 +36,9 @@ func IsBucketNotFound(err error) bool {
 	if strings.Contains(aerr.Code(), "Not Found") {
 		return true
 	}
+	if microerror.Cause(err) == notFoundError {
```

Comment:
The cause should also be used to assert the AWS error in the very first line. Also, the not found error has nothing to do here. There is already an assertion for the not found error. The matcher here is for buckets not being found and thus there should be a specific error for this which we assert here. We always have an error and a matcher for it. Lets please not mix things up and abuse in different places.

Reply:
The problem here is the expression `microerror.Cause(aerr)` returns `"NotFound: Not Found\n\tstatus code: 404, request id: 00B7939985F4AE19, host id: Vh7D62xqbTSorifIzGY+2xbdnpPE2ahRH2CFIHoI+YnMzlubk7BGiolsfswZTj2USweivAUqDuQ="%`  I think the AWS API does not return the correct response here. It is the reason I used `Contains` method. Any idea how to solve it in a prettier way?